controllers/volumereplication_controller.go

```diff
@@ -317,6 +325,64 @@ func (r *VolumeReplicationReconciler) SetupWithManager(mgr ctrl.Manager, cfg *co
 		WithEventFilter(pred).Complete(r)
 }
 
+func (r *VolumeReplicationReconciler) waitForCrds() error {
+
+	logger := r.Log.WithName("Wait for VolumeReplicationClass and VolumeReplication crds")
+
+	err := r.waitForVolumeReplicationClass(logger)
+	if err != nil {
+		logger.Error(err, "Failed to watch VolumeReplicationClass")
+		return err
+	}
+
+	err = r.waitForVolumeReplication(logger)
+	if err != nil {
+		logger.Error(err, "Failed to watch VolumeReplication")
+		return err
+	}
+	return nil
+}
+
+func (r *VolumeReplicationReconciler) waitForVolumeReplicationClass(logger logr.Logger) error {
+
+	vrcObj := &replicationv1alpha1.VolumeReplicationClass{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: ""}, vrcObj)
+	if err != nil {
+		if meta.IsNoMatchError(err){
+			logger.Info("Resource", "VolumeReplicationClass", "does not exist")
+			time.Sleep(5 * time.Second)
+			err = r.waitForVolumeReplicationClass(logger)
```

Comment:
Instead of recursion we can just use a for(ever) loop here and break on errors or if the CRD is found?

I do not know ho recursion in go works, but this will accumulate a function on the stack every 5 seconds, and for a sidecar that expects a CRD to come at some point in time in the future, this maybe quite a lot of accumulations.

(applies to `waitForVolumeReplication` as well)

Reply:
done